Source: dleverton
Upstream: http://sourceforge.net/tracker/index.php?func=detail&aid=1914985&group_id=66144&atid=513479
Reason: fix EFAULT when calling the preprocessor on 64bit systems
        (only seems to trigger when grcompiler-r829-include-path.patch
        is also applied /and/ the grcompiler binary is prelinked, but
        only by luck, since the code is definitely incorrect)

Index: grcompiler/compiler/ParserTreeWalker.cpp
===================================================================
--- grcompiler/compiler/ParserTreeWalker.cpp	(revision 829)
+++ grcompiler/compiler/ParserTreeWalker.cpp	(working copy)
@@ -213,9 +213,9 @@
 		exit(-1);
 		case 0 : 
 			if (m_verbose)
-				testexec = execlp(staGdlppFile.c_str(),staGdlppFile.c_str(),"-V",staFileName.c_str(),tmpgdl,0); // this is the code the child runs
+				testexec = execlp(staGdlppFile.c_str(),staGdlppFile.c_str(),"-V",staFileName.c_str(),tmpgdl,static_cast<const char *>(0)); // this is the code the child runs
 			else
-				testexec = execlp(staGdlppFile.c_str(),staGdlppFile.c_str(),staFileName.c_str(),tmpgdl,0); // this is 
+				testexec = execlp(staGdlppFile.c_str(),staGdlppFile.c_str(),staFileName.c_str(),tmpgdl,static_cast<const char *>(0)); // this is 
 			cout << "// exec retval:" << testexec << ", errno:" << strerror(errno) << "(" << errno << ")\n";
 			cout << "// tmpfile " << tmpgdl << endl;
 			cout << "// file " << staFileName.c_str() << endl;
